linux_boot_params/src/lib.rs

```diff
@@ -609,6 +626,26 @@ pub struct BootParams {
 }
 static_assertions::assert_eq_size!(BootParams, [u8; 4096usize]);
 
+impl BootParams {
+    pub fn protocol(&self) -> &'static str {
+        "Linux Boot Protocol"
+    }
+
+    pub fn e820_table(&self) -> &[BootE820Entry] {
```

Comment:
What's an E820 table?

Reply:
It refers to a BIOS function that provides the memory layout to the operating system:

https://en.wikipedia.org/wiki/E820
https://wiki.osdev.org/Detecting_Memory_(x86)#BIOS_Function:_INT_0x15.2C_EAX_.3D_0xE820

Originally invoked by setting `EAX` to `0xE820` (hence the name) and then invoking interrupt 15h, we don't really have a BIOS API available for us, but the name has stuck for this kind of "memory map" data structures. Hence the name here.

Otherwise, the table contains information about which physical memory address ranges are actually valid (not every physical address has a RAM chip behind it, after all), which ranges are reserved, and so on.